borg/helpers.py

```diff
@@ -699,6 +705,10 @@ class Buffer:
     """
     provide a thread-local buffer
     """
+
+    class OutOfMemory(Error, OSError):
```

Comment:
btw, this class name is a bit misleading: we are not out of memory, we just reject to go beyond that as we suspect there is some malfunction if we get this far. BufferSizeLimitExceeded?